aopy/preproc.py

```diff
@@ -366,14 +367,24 @@ def trial_separate(events, times, evt_start, n_events=8):
     trial_events = np.empty((num_trials, n_events), dtype=events.dtype)
     trial_times = np.empty((num_trials, n_events), dtype=times.dtype)
     for iE in range(len(evt_start_idx)):
-        trial_events[iE,:] = events[evt_start_idx[iE]: evt_start_idx[iE]+n_events]
-        trial_times[iE,:] = times[evt_start_idx[iE]: evt_start_idx[iE]+n_events]
+
+        if len(events[evt_start_idx[iE]: evt_start_idx[iE]+n_events]) == n_events:
+            trial_events[iE,:] = events[evt_start_idx[iE]: evt_start_idx[iE]+n_events]
+            trial_times[iE,:] = times[evt_start_idx[iE]: evt_start_idx[iE]+n_events]
+        else:
+            short_event_len = len(events[evt_start_idx[iE]: evt_start_idx[iE]+n_events])
+            trial_events[iE,:short_event_len] = events[evt_start_idx[iE]: evt_start_idx[iE]+short_event_len]
+            trial_events[iE, short_event_len:] = 0
+            trial_times[iE,:short_event_len] = times[evt_start_idx[iE]: evt_start_idx[iE]+short_event_len]
+            trial_times[iE,short_event_len:] = 0
+
```

Comment:
i think this works for integer events but it won't work for string events. i'll make a commit with some suggested changes.